statsmodels/tsa/statespace/structural.py

```diff
@@ -203,6 +214,39 @@ class UnobservedComponents(MLEModel):
     time series is available in the results class in the `seasonal`
     attribute.
 
+    ** Freq. Seasonal**
+
+    Each frequency-domain seasonal component is modeled as:
+
+    .. math::
+        \gamma_t &=& \sum_{j=1}^h \gamma_{j, t}
+        \gamma_{j, t+1} &=& \gamma_{j, t}\cos(\lambda_j) \\
+                        && + \gamma^{*}_{j, t}\sin(\lambda_j) + \omega_{j,t} \\
+        \gamma^{*}_{j, t+1} &=^ -\gamma^{(1)}_{j, t}\sin(\lambda_j) \\
+                            &&+ \gamma^{*}_{j, t}\cos(\lambda_j) \\
+                            + \omega^{*}_{j, t}, \\
+        \omega^{*}_{j, t}, \omega_{j, t} &\sim& N(0, \sigma_{\omega^2}
+        \lambda_j = \frac{2 \pi j}{s}
+    where j ranges from 1 to h.
+
+    The periodicity (number of "seasons" in a "year") is s and the number of
+    harmonics is h.  Note that h is configurable to be less than s/2, but
+    s/2 harmonics is sufficient to fully model all seasonal variations of
```

Comment:
Is it better to say "s/2 harmonics is sufficient" or "s/2 harmonics are required"? IMO the first makes it seem like you might sometimes use more, while I think your point is that you might sometimes use less for parsimony but if you want to model all seasonal variations then you need s/2?

Am I understanding correctly?

Reply:
Yes you have my main point.  I was also thinking that for artificial seasonal components (i.e. nature is a single sin wave), "required" isn't correct.  I like your phrasing better though, since virtually any "real" periodic function will not be perfectly modeled by less than s/2 trig. pairs.